### PR TITLE
GROOVY-9546: groovydoc: fix annotation rendering

### PR DESCRIPTION
**Pull request by vmj**
First stab at fixing the annotation rendering.

Somehow the annotations are not shown on properties at all.

Let me know if this is too hacky.

### CODE REVIEW

**Comment by vmj** [comment]
Created a separate issue for the missing property annotations, since I couldn't yet figure it out: https://issues.apache.org/jira/browse/GROOVY-9548

File: subprojects/groovy-groovydoc/src/test/groovy/org/codehaus/groovy/tools/groovydoc/testfiles/anno/Groovy.groovy [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *  under the License.
+ */
+package org.codehaus.groovy.tools.groovydoc.testfiles.anno
+
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.NamedParam
+import groovy.transform.NamedVariant
+
+// The annotations that are used here don't really matter,
+// since groovydoc shows all annotations, not just @Documented ones.
```
Comment (paulk-asert) [inline subprojects/groovy-groovydoc/src/test/groovy/org/codehaus/groovy/tools/groovydoc/testfiles/anno/Groovy.groovy:26]: We do plan to correct this one day. We currently try to minimise resolving classes when not needed but we will likely change that at some point.

**Comment by paulk-asert** [comment]
Merged. Thanks!

File: subprojects/groovy-groovydoc/src/test/groovy/org/codehaus/groovy/tools/groovydoc/testfiles/anno/Groovy.groovy (hunk 1 above)
Comment (vmj) [reply subprojects/groovy-groovydoc/src/test/groovy/org/codehaus/groovy/tools/groovydoc/testfiles/anno/Groovy.groovy:26]: Got it!  [GROOVY-4634](https://issues.apache.org/jira/browse/GROOVY-4634) is probably it, then.

Comment (paulk-asert) [reply subprojects/groovy-groovydoc/src/test/groovy/org/codehaus/groovy/tools/groovydoc/testfiles/anno/Groovy.groovy:26]: That's the one. :-)

**Pull request closed without merging**